### ISSUE TITLE
dbt docs generate should also invoke compilation

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#932]
### Feature description
`dbt docs generate` should invoke the same code path as `dbt compile`. This is important for the documentation use case, where the `run_results.json` asset is used to show compiled SQL in the docs site. If a user runs:

```
dbt run --model one_specific_model
dbt docs generate
dbt docs serve
```

then they'll only see the compiled sql for `one_specific_model`.  Instead, `dbt docs generate` should invoke compilation and produce a `run_results.json` asset with an entry for every node in the dag.

**Comment by cmcarthur** [comment]
Implement `--no-compile` flag to skip compilation

**Comment by beckjake** [comment]
Fixed in #936
